centreon/src/Core/Broker/Application/Exception/BrokerException.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Broker\Application\Exception;
+
+class BrokerException extends \Exception
+{
+    public const CODE_CONFLICT = 1;
+
+    /**
+     * @return self
+     */
+    public static function addBrokerOutput(): self
+    {
+        return new self(_('Error while adding a broker output'));
+    }
+
+    /**
+     * @return self
+     */
+    public static function editNotAllowed(): self
+    {
+        return new self(_('You are not allowed to edit a broker configuration'));
+    }
+
+    /**
+     * @param int $brokerId
+     *
+     * @return self
+     */
+    public static function notFound(int $brokerId): self
+    {
+        return new self(sprintf(_('Broker configuration #%d not found'), $brokerId));
+    }
+
+    /**
+     * @param string $propertyName
+     * @param int $propertyValue
+     *
+     * @return self
+     */
+    public static function idDoesNotExist(string $propertyName, int $propertyValue): self
+    {
+        return new self(
+            sprintf(_("The %s with value '%d' does not exist"), $propertyName, $propertyValue),
+            self::CODE_CONFLICT
```

Comment:
For this one and other CODE_CONFLICT, I'm not sure this should be the good code, all the 4 error with CODE_CONFLICT seems due to invalid input from the request, i think a CODE_INVALID or any other convenient name should be better (then in the useCase return a 400 instead of a 409)